src/lib/tuning/tuning_choice.cpp

```diff
@@ -0,0 +1,69 @@
+#include "tuning_choice.hpp"
+
+#include "null_operation.hpp"
+
+namespace opossum {
+
+float TuningChoice::accept_desirability() const {
+  if (!is_currently_chosen()) {
+    return desirability();
+  }
+  return 0.0f;
+}
+
+float TuningChoice::reject_desirability() const {
+  if (is_currently_chosen()) {
+    return -desirability();
```

Comment:
Just until now, I was under the impression that accept and reject desirability were two different things. Having `desirability()` as a third public method makes this even more complicated. Again, this highlights the need to find better names.

Reply:
`desirability()` is independent of the system state and the `accept_*`, `reject_*` and `current_*` methods are state-dependent methods that are intended to be used by a Selector decide whether to accept or reject a choice. This is documented in detail in `tuning_choice.hpp`, but if you like a further distinction, we might sacrifice conciseness and add `_delta` to the helpers, like `reject_desirability_delta()` (delta meaning: What happens to the desirability balance if the choice is accepted or rejected)